### PR TITLE
Add html validation to radio buttons on category tree

### LINKED ISSUE #19612: CategoryChoiceTreeType, ChoiceCategoriesTreeType ShopChoiceTreeType "required" attibute doesn't validate fields
#### Describe the bug

When you use a Material Tree Type from PrestaShopBundle, you can set required to true.

You can see the red star at the right if the label, but you can validate. There is no HTML5 validation

image

#### Expected behavior

Manage HTML5 validation.

#### Steps to Reproduce

1. Install PS 1.7.7 and [sample module](https://github.com/202ecommerce/checkboform)
2. See error by validating the form

#### Additional information

* PrestaShop version: 1.7.6 / 1.7.7
* PHP version: N/A

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Category trees are not blocking the submit form if required
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #19612.
| How to test?      | See issue, btw, it doesn't fix the material checkbox one, that should be done using JavaScript inside the UIKit, with an helper class in order to create the control : 1. Install PS and [sample module (https://github.com/202ecommerce/checkboform) 2. See error by validating the form without checking checkboxes (won't work on material checkboxes)
| Possible impacts? | Every radio trees

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22942)

⚠️ Mergeable when https://github.com/PrestaShop/prestashop-ui-kit/pull/137 is merged and UIKit 1.2.2 is released ⚠️

### CODE REVIEW

**Review by matks — DISMISSED** [review]
Code seems good but could you please make it a little more clear what is fixed, what is not, and how QA can validate this PR 😉

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

I have only one issue in the category page when multistore enabled.
Steps to reproduce the issue:
1. Enable Multistore Context
2. Create a new shop
3. Go to BO > Catalog > Categories page
4. Add a new root category > Save 
5. At this step, you are in the Root category list 
image
6. Add a new category > the Parent category is not selected
7. Set only the Name & save
8. See the error => nothing happens

I attached a screen record
https://drive.google.com/file/d/11PXqvCzvJq0lrAcG4DW0eRn97kjfoUFE/view?usp=sharing

**Without PR**
When I click save => I have an alert displayed
image

Thank you!

**Comment by NeOMakinG** [comment]
@khouloudbelguith 

Thanks for testing, this error isn't related at all, it's just because the form doesn't submit as you didn't fill the category tree, btw I shouldn't block these fields as they can't be fixed only by doing this, I need to go further on this PR.

Also, I don't understand why the submit isn't done and your field isn't required, maybe I miss something

**Comment by NeOMakinG** [comment]
@khouloudbelguith could you test again ? I checkout the PR and it looks like I can't reproduce your behavior at all !

Edit: I reproduced it and it's related to the PR

**Comment by NeOMakinG** [comment]
I added the "required" little asterisk and fixed this problem on UIKit side : https://github.com/PrestaShop/prestashop-ui-kit/pull/137

It's testable with this PR as well

File: src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig [hunk 1]
```diff
@@ -200,7 +201,7 @@
         {%- else -%}
             <div class="radio">
                 <label>
-                    <input type="radio" name="form[{{ form.vars.id }}][tree]" value="{{ child.id_category }}" {{ checked }} class="category">
+                    <input type="radio" name="form[{{ form.vars.id }}][tree]" value="{{ child.id_category }}" {{ checked }} class="category" {% if required %}required{% endif %}>
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig]:
```suggestion
                    <input type="radio" name="form[{{ form.vars.id }}][tree]" value="{{ child.id_category }}" {{ checked }} class="category"{% if required %} required{% endif %}>
```

File: src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/material.html.twig [hunk 2]
```diff
@@ -24,7 +24,7 @@
  *#}
 
 {% block material_choice_tree_widget %}
-<div class="material-choice-tree-container js-choice-tree-container" id="{{ form.vars.id }}">
+  <div class="material-choice-tree-container js-choice-tree-container {% if required %}required{% endif %}" id="{{ form.vars.id }}">
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/material.html.twig]:
```suggestion
  <div class="material-choice-tree-container js-choice-tree-container{% if required %} required{% endif %}" id="{{ form.vars.id }}">
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Just two comments :)

File: src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig [hunk 3]
```diff
@@ -186,7 +186,8 @@
         {% if multiple -%}
             <div class="checkbox">
                 <label>
-                    <input type="checkbox" name="{{ form.vars.full_name }}[tree][]" value="{{ child.id_category }}" class="category" {{ checked }}>
+                  <input type="checkbox" name="{{ form.vars.full_name }}[tree][]" value="{{ child.id_category }}" class="category" {{ checked }}>
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/TwigTemplateForm/bootstrap_4_layout.html.twig]:
```suggestion
                    <input type="checkbox" name="{{ form.vars.full_name }}[tree][]" value="{{ child.id_category }}" class="category" {{ checked }}>
```

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

Using this PR: https://github.com/PrestaShop/prestashop-ui-kit/pull/137, I found only an error in the console
Steps to reproduce the issue:

1. Enable Multistore Context
2. Create a new shop
3. Go to BO > Catalog > Categories page
4. Add a new root category > Save
5. At this step, you are in the Root category list
6. Select all only the `first shop`/`second shop`
7. I have an error but it is only on the console
8. Add a new category > the Parent category is not selected
9. Set only the Name & save
10. See the error on the console

image

What do you think?

Thank you!

**Comment by NeOMakinG** [comment]
@khouloudbelguith 

Thanks a lot for testing.

I understand this error, it's due to radio elements which are inside a `display: none;` property. so the browser is unable to show the error message on an hidden element.

I think it's fine enough for this work, overwise it would need a lot of javascript in order to handle every forms, which could lead to side effects. Or a rework of radio buttons which is a BC...

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — COMMENTED** [review]
Please fix review from @PierreRambaud

**Comment by NeOMakinG** [comment]
@Progi1984 idk why it has been removed, but I fixed it again :)

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

An issue is created here: https://github.com/PrestaShop/PrestaShop/issues/23365

It is ok :heavy_check_mark: 

**With PR**
image
**Without PR**
image

Thank you!

**Review by Progi1984 — CHANGES_REQUESTED** [review]
## WAITING FOR RELEASE OF UIKit 1.2.2

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @khouloudbelguith

**Pull request merged**